### PR TITLE
Fixes sinks giving misleading messages when containers full

### LINKED ISSUE #24879: Infinite filling of bucket & water backpack
You keep getting messages like "You fill the bucket from the sink." when filling the botany water backpack or the bucket. There should be a limit of water that the container can hold and it would let you know when it's full.

### PR DESCRIPTION
**Pull request by optimumtact**
Sinks will now give a message when the reagent holder is full

This also adds a helpful proc to check a reagent holder is full

Fixes #24879

### CODE REVIEW

**Comment by Supermichael777** [comment]
Memes

**Comment by PKPenguin321** [comment]
Misleading title rreeeeeeee!!!!!!

:+1:

**Comment by Niknakflak** [comment]
This is a violation of the misleading title policy

**Comment by optimumtact** [comment]
Look at me, I'm the maintainer

**Comment by Kraseo** [comment]
stupid kiwi go to BED

File: code/game/objects/structures/watercloset.dm [hunk 1]
```diff
@@ -456,9 +456,12 @@
...
 		if(RG.container_type & OPENCONTAINER)
-			RG.reagents.add_reagent("[dispensedreagent]", min(RG.volume - RG.reagents.total_volume, RG.amount_per_transfer_from_this))
-			to_chat(user, "<span class='notice'>You fill [RG] from [src].</span>")
-			return 1
+			if(!RG.reagents.holder_full())
+				RG.reagents.add_reagent("[dispensedreagent]", min(RG.volume - RG.reagents.total_volume, RG.amount_per_transfer_from_this))
+				to_chat(user, "<span class='notice'>You fill [RG] from [src].</span>")
+				return TRUE
+			to_chat(user, "<span class='notice'>The [RG] is full.</span>")
+			return FALSE
```
Comment (MrPerson) [inline code/game/objects/structures/watercloset.dm]: Won't this output "The the sink is full."?

Comment (optimumtact) [inline code/game/objects/structures/watercloset.dm]: no?

Comment (MrPerson) [inline code/game/objects/structures/watercloset.dm]: Sorry I'm an idiot, I meant "The the beaker is full."

Comment (RemieRichards) [inline code/game/objects/structures/watercloset.dm]: it should be ```\The```

**Pull request merged**